scan/lib/scan/runner.rb

```diff
@@ -80,8 +84,16 @@ def handle_results(tests_exit_status)
     end
 
     def test_results
-      return "" unless Scan.cache[:temp_junit_report]
-      File.read(Scan.cache[:temp_junit_report])
+      temp_junit_report = Scan.cache[:temp_junit_report]
+      return File.read(temp_junit_report) if temp_junit_report && File.file?(temp_junit_report)
```

Comment:
1. yes. For now :)
2. for troubleshooting GC, we could try adding extra finalizers `ObjectSpace.define_finalizer(a, proc {|id| puts "Finalizer one on #{id}" })`
3. alternatively we could let the Runner create the temp file and pass it down to the call stack (or pass an object whose responsibility is to allocate the Tempfile). But the lifecyle of that object would be explicit in the run function for example. And no need to make all those other changes.